pkg/snapstore/utils.go

```diff
@@ -169,3 +170,29 @@ func adaptPrefix(snap *brtypes.Snapshot, snapstorePrefix string) string {
 
 	return snapstorePrefix
 }
+
+func GetSnapstoreSecretHash(config *brtypes.SnapstoreConfig) (string, error) {
+	switch config.Provider {
+	case brtypes.SnapstoreProviderLocal:
+		return "", nil
+	case brtypes.SnapstoreProviderS3:
+		return S3SnapStoreHash(config)
+	case brtypes.SnapstoreProviderABS:
+		return ABSSnapStoreHash(config)
+	// case brtypes.SnapstoreProviderGCS:
+	// 	return GCSSnapStoreHash(config)
+	case brtypes.SnapstoreProviderSwift:
+		return SwiftSnapStoreHash(config)
+	case brtypes.SnapstoreProviderOSS:
+		return OSSSnapStoreHash(config)
+	default:
+		return "", nil
+	}
+}
+
+func getHash(data string) string {
+	hash := sha256.New()
+	hash.Write([]byte(data))
+	sha := hash.Sum(nil)
```

Comment:
sha256 is a pkg name. We need obj to call Sum.
```
func (d *digest) Sum(in []byte) []byte {
	// Make a copy of d so that caller can keep writing and summing.
	d0 := *d
	hash := d0.checkSum()
	if d0.is224 {
		return append(in, hash[:Size224]...)
	}
	return append(in, hash[:]...)
}
```

Reply:
It could be something like this `sha256.New().Sum([]byte(data))`

Reply:
Sorry, I meant `sha256.Sum256(data)` but `sha256.New().Sum([]byte(data))` is also fine :)